test/parallel/test-runner-watch-mode.mjs

```diff
@@ -0,0 +1,52 @@
+import '../common/index.mjs';
+import { describe, it } from 'node:test';
+import { spawn } from 'node:child_process';
+import { writeFileSync } from 'node:fs';
+import * as fixtures from '../common/fixtures.mjs';
+
+function deferred() {
+  let res;
+  const promise = new Promise((resolve) => res = resolve);
+  return { resolve: res, promise };
+}
+
+
+describe('test runner watch mode', { concurrency: false }, () => {
+  it('should run tests repeatedly', async () => {
+    const ran1 = deferred();
+    const ran2 = deferred();
+    const file1 = fixtures.path('test-runner/index.test.js');
+    const file2 = fixtures.path('test-runner/subdir/subdir_test.js');
+    const child = spawn(process.execPath, ['--watch', '--test', '--no-warnings', file1, file2], { encoding: 'utf8' });
+    child.stdout.on('data', (data) => {
+      const num = /ok (\d+) -/.exec(data.toString())?.[1];
+      if (num === '2') ran1.resolve();
+      if (num === '3') ran2.resolve();
+    });
+
+    await ran1.promise;
+    writeFileSync(file2, '');
+    await ran2.promise;
+    child.kill();
+  });
+
+  it('should run tests with dependency repeatedly', async () => {
+    const ran1 = deferred();
+    const ran2 = deferred();
+    const file1 = fixtures.path('test-runner/index.test.js');
+    const dependant = fixtures.path('test-runner/dependant.js');
+    const dependancy = fixtures.path('test-runner/dependency.js');
+    const child = spawn(process.execPath,
+                        ['--watch', '--test', '--no-warnings', file1, dependant], { encoding: 'utf8' });
+    child.stdout.on('data', (data) => {
+      const num = /ok (\d+) -/.exec(data.toString())?.[1];
```

Comment:
Same comment here. If the test point ever gets split across chunks this test will be flaky.